app/merginapi.h

```diff
@@ -5,37 +5,75 @@
 #include <QNetworkAccessManager>
 #include <QEventLoop>
 #include <memory>
+#include <QFile>
+
+enum ProjectStatus
+{
+  NoVersion = Qt::UserRole + 1,
+  Pending,
+  UpToDate,
+  OutOfDate
+};
+Q_ENUMS( ProjectStatus )
 
 struct MerginProject {
     QString name;
     QStringList tags;
     QString info;
+    bool pending = false;
+    ProjectStatus status = NoVersion;
+};
+
+struct MerginFile {
+    QString path;
+    QString checksum;
 };
+
+
 typedef QList<std::shared_ptr<MerginProject>> ProjectList;
 
 class MerginApi: public QObject {
-     Q_OBJECT
+    Q_OBJECT
 public:
-    explicit MerginApi(const QString& root, QByteArray token, QObject* parent = nullptr );
+    explicit MerginApi(const QString& root, const QString& dataDir, QByteArray token, QObject* parent = nullptr );
     ~MerginApi() = default;
     Q_INVOKABLE void listProjects();
+    Q_INVOKABLE void downloadProject(QString projectName);
+    Q_INVOKABLE void getInfoAndFetch(QString projectName);
```

Comment:
can you please add some docs on the above three methods - what they do, that they are non-blocking and what signals to expect in the client code...